simulator/modules/fetch/bpu/t/unit_test.cpp

```diff
@@ -26,16 +26,16 @@ TEST_CASE( "Static, all branches not taken")
     CHECK( bp->get_target(PC) == target);
 }
 
-//TEST_CASE( "Static, all branches taken")
-//{
-//    auto bp = BaseBP::create_bp( "always_taken", 128, 16);
-//
-//    Addr PC = 28;
-//    Addr target = 12;
-//    
-//    CHECK( bp->is_taken(PC) );
-//    CHECK( bp->get_target(PC) == target);
-//}
+TEST_CASE( "Static, all branches taken")
+{
+    auto bp = BaseBP::create_bp( "always_taken", 128, 16);
+
+    Addr PC = 28;
+    Addr target = 12;
+    
+    //CHECK( bp->is_taken(PC) );
```

Comment:
That one passes, right?
Do you run unit tests locally?

Reply:
No, it doesn't as you wrote to me
```
1: /home/travis/build/MIPT-ILab/mipt-mips/simulator/modules/fetch/bpu/t/unit_test.cpp:36: FAILED:
1:   CHECK( bp->is_taken(PC) )
1: with expansion:
1:   false
```

Reply:
Oh, sorry, my fault.

Reply:
I have a problem with unit tests running locally, I have a lot of tests on caches failed. I don't know why, I have updated my repo and rebuilt typing
```
cmake CMakeLists.txt
make
```
but still I have failed tests

Reply:
For example
```/home/yan/mipt-mips/simulator/infra/cache/t/unit_test.cpp:100: FAILED:
  CHECK( miss == sample_miss )
with expansion:
  30400 (0x76c0)
  ==
  139705591643040 (0x7f0fbe29bba0)

/home/yan/mipt-mips/simulator/infra/cache/t/unit_test.cpp:99: FAILED:
  CHECK( hit == sample_hit )
with expansion:
  1088685 (0x109cad) == 0

/home/yan/mipt-mips/simulator/infra/cache/t/unit_test.cpp:100: FAILED:
  CHECK( miss == sample_miss )
with expansion:
  30400 (0x76c0)
  ==
  139705591643040 (0x7f0fbe29bba0)
```

Reply:
All right then, in that case commenting full test case was the right idea, as compilers blame for unused variables.

Reply:
> /home/yan/mipt-mips/simulator/infra/cache/t/unit_test.cpp:99: FAILED:

There are no tests in that line of the repository anymore:
https://github.com/MIPT-ILab/mipt-mips/blob/837d58eb992d7ed5175ee0e639c2916f057cf779/simulator/infra/cache/t/unit_test.cpp#L99

Could you please try to build in another directory?

Reply:
I resolved it typing 
```
make unit-tests
```
So, before it I simply had not actual unit tests